binders/kafka-binder/spring-cloud-stream-binder-kafka-reactive/src/main/java/org/springframework/cloud/stream/binder/reactorkafka/ReactorKafkaBinder.java

```diff
@@ -169,7 +170,13 @@ protected MessageHandler createProducerMessageHandler(ProducerDestination destin
 				SenderOptions.create(configs));
 		// TODO bean for converter; MCB doesn't use one on the producer side.
 		RecordMessageConverter converter = new MessagingMessageConverter();
-		return new ReactorMessageHandler(opts, converter, destination.getName());
+		AbstractApplicationContext applicationContext = getApplicationContext();
+		FluxMessageChannel resultChannel = null;
+		String channelName = producerProperties.getExtension().getRecordMetadataChannel();
+		if (channelName != null && applicationContext.containsBean(channelName)) {
+			resultChannel = applicationContext.getBean(channelName, FluxMessageChannel.class);
```

Comment:
The `KafkaProducerProperties.recordMetadataChannel` must reflect this requirement. Kinda just enough to say `The bean name of a FluxMessageChannel...`